Aspire Budgeting/Views/FileSelectorView.swift

```diff
@@ -2,57 +2,67 @@
+  }
 
-  @State var selectedFile: File?
+  var error: Error? {
+    viewModel.getError()
+  }
+
+  @State private var searchText = ""
 
   var body: some View {
-    ZStack {
-      if self.userManager.user != nil {
-        ZStack {
-          if self.selectedFile == nil {
-            NavigationView {
-              List(self.driveManager.fileList) { file in
-                Button(
-                  action: {
-                    self.sheetsManager.defaultFile = file
-                    self.selectedFile = file
-                  }, label: {
-                    Text(file.name)
-                  }
-                )
-              }
-              .navigationBarTitle("Link your Aspire sheet")
-            }.onAppear {
-              if self.driveManager.fileList.isEmpty {
-                self.driveManager.getFileList()
+    if viewModel.currentState == .isLoading {
+      Text("Loading")
+    }
+
+    if viewModel.currentState == .filesRetrieved {
+      NavigationView {
+        VStack {
+          SearchBar(text: $searchText)
+          List(files.filter({searchText.isEmpty ? true : $0.name.contains(searchText)})) { file in
+            Button(
+              action: {
+                viewModel.fileSelectedCallback?(file)
+              }, label: {
+                HStack {
+                  Image("sheetsIcon").renderingMode(.original)
+                  Text(file.name).font(.nunitoBold(size: 16))
+                }.frame(height: 60)
               }
-            }
-          }
-          if self.selectedFile != nil {
-            AspireMasterView()
-          }
-        }
-      }
-      if self.driveManager.error != nil {
-        Text("Error occured")
+            )
+          }.padding(.bottom, 20).navigationBarTitle("Link your Aspire sheet")
+        }.background(Color.primaryBackgroundColor.edgesIgnoringSafeArea(.all))
       }
     }
+
+    if viewModel.currentState == .error {
+      Text("Error Occured: \(error?.localizedDescription ?? "")")
+    }
   }
 }
 
-// struct FileSelectorView_Previews: PreviewProvider {
-//    static var previews: some View {
-//        FileSelectorView()
-//    }
-// }
+ struct FileSelectorView_Previews: PreviewProvider {
+  static let files = [File(id: "abc", name: "File 1"),
+                      File(id: "def", name: "File 2"),
+  ]
+    static var previews: some View {
+      Group {
+        FileSelectorView(viewModel: FileSelectorViewModel(fileManagerState:
+                                                            .filesRetrieved(files: FileSelectorView_Previews.files),
+                                                          fileSelectedCallback: nil))
+
+        FileSelectorView(viewModel: FileSelectorViewModel(fileManagerState:
+                                                            .filesRetrieved(files: FileSelectorView_Previews.files),
+                                                          fileSelectedCallback: nil)).environment(\.colorScheme, .dark)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/ff44934e95fcfbf9892d0e49040b4320aa9b4509/Aspire Budgeting/Views/FileSelectorView.swift#L65"></span>Line should be 100 characters or less: currently 119 characters<br />
<code>line_length</code> <code>FileSelectorView.swift:65</code></td>
    </tr>
  </tbody>
</table>